### ISSUE TITLE
devices list is empty

### ISSUE THREAD

**Issue by macphailmagwira** [andrey-ushakov/esc_pos_bluetooth#49]
I'm testing to see if the plugin can detect nearby Bluetooth devices but the device list is empty

**Comment by agtt** [comment]
Only works on real device and does not find all Bluetooth devices only bounded devices.

**Comment by ziakhan110** [comment]
You need to turn on GPS to discover devices.

**Comment by abnerh69** [comment]
GPS?

> You need to turn on GPS to discover devices.

**Comment by dihsarjr** [comment]
Same error for me

**Comment by von-maurus** [comment]
Same error here, I have these statements in AndroidManifest

`<uses-permission android:name="android.permission.INTERNET"/>
    <uses-permission android:name="android.permission.BLUETOOTH" />
    <uses-permission android:name="android.permission.BLUETOOTH_ADMIN" />
    <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION"/>
    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION"/>
    <uses-permission android:name="android.permission.ACCESS_BACKGROUND_LOCATION"/>`

The list of devices is empty all the time.  Also I have GPS activated in the smartphone. 
My build.gradle has a 
`defaultConfig {
        // TODO: Specify your own unique Application ID (https://developer.android.com/studio/build/application-id.html).
        applicationId "ID"
        minSdkVersion 21
        targetSdkVersion 29
        versionCode flutterVersionCode.toInteger()
        versionName flutterVersionName
    }`

Can someone help?

**Comment by ozhankucuk** [comment]
Hi
I have same error but I fixed .
A added below peermission Manifest.xml

<uses-permission android:name="android.permission.ACCESS_BACKGROUND_LOCATION" /> 
then When I opened select printer page , I will check permission and then all printers will be shown.
await Permission.locationAlways.request().isGranted;

printerManager.scanResults.listen((devices) async {
  // print('UI: Devices found ${devices.length}');
  setState(() {
    _devices = devices;
  });
});
_startScanDevices();

**Comment by harshilishere** [comment]
Tried the permissions as mentioned by @von-maurus but still nothing. I can't see anything on the list. Please someone help.

**Comment by von-maurus** [comment]
> Tried the permissions as mentioned by @von-maurus but still nothing. I can't see anything on the list. Please someone help.

@harshilishere Hi, I think I fixed this by activating my GPS location on the smarthphone

**Comment by harshilishere** [comment]
@von-maurus I tried that but it didn't happened.

**Comment by ziakhan110** [comment]
Yes you need GPS(Location services) enabled and your app must have location permission to access Bluetooth devices on android 10 and above

**Comment by von-maurus** [comment]
> @von-maurus I tried that but it didn't happened.

Can you post some code or exception?
